packages/interpreter/src/unified_bindings.rs

```diff
@@ -74,16 +74,30 @@ mod js {
     fn create_placeholder(id: u32) {
         "{let node = document.createElement('pre'); node.hidden = true; this.stack.push(node); this.nodes[$id$] = node;}"
     }
+
     fn new_event_listener(event_name: &str<u8, evt>, id: u32, bubbles: u8) {
-        r#"
-            let node = this.nodes[id];
-            if(node.listening){node.listening += 1;}else{node.listening = 1;}
+        r#"{
+            const event_name = $event_name$;
```

Comment:
This is a hot path, so we do some extra work for optimization. The `$$`'s mean the parameter is only used once and it is safe to inline the decoding logic into where the parameter is used